tls/s2n_security_policies.c

```diff
@@ -1459,3 +1459,34 @@ S2N_RESULT s2n_security_policy_get_version(const struct s2n_security_policy *sec
     }
     RESULT_BAIL(S2N_ERR_INVALID_SECURITY_POLICY);
 }
+
+S2N_RESULT s2n_security_policy_validate_sig_scheme_supported(const struct s2n_cert_info *info,
+        const struct s2n_signature_preferences *cert_sig_preferences)
+{
+    RESULT_ENSURE_REF(info);
+    RESULT_ENSURE_REF(cert_sig_preferences);
+
+    for (size_t i = 0; i < cert_sig_preferences->count; i++) {
+        if (cert_sig_preferences->signature_schemes[i]->libcrypto_nid == info->signature_nid) {
+            return S2N_RESULT_OK;
+        }
+    }
+
+    RESULT_BAIL(S2N_ERR_CERT_UNTRUSTED);
+}
+
+S2N_RESULT s2n_security_policy_validate_certificate(const struct s2n_cert_info *info,
+        const struct s2n_security_policy *security_policy)
+{
+    RESULT_ENSURE_REF(info);
+    RESULT_ENSURE_REF(security_policy);
+
+    if (info->self_signed) {
```

Comment:
I'm curious why this check needs to be its own function rather than combined with s2n_security_policy_validate_sig_scheme_supported or s2n_validator_check_cert_preferences?

Reply:
It's not immediately clear to me that we need that check to apply to local certificates. Or at least that's a discussion I could see us having.